### PR TITLE
Added site information to CI and `alpha` job.

### PR DESCRIPTION
**Pull request by AlexSkrypnyk**
**TL;DR Our D11 CI is not in line with D10 CI leading to a coverage gap**

The test jobs across D10 and D11 should be handling the same 3 types of releases: `current`, `next` and `canary`. D10 already does this, but D11 does not.

For a different Drupal version at a specific point in time Composer will resolve packages versions according to specific stability available at that time. We set that stability resolution as `minimum-stability` in `composer.json`.

For D10, we currently testing exactly that.

For D11 we currently handle "current" as "next" and is missing what is currently set in `minimum-stability` (the value is `alpha`) within a `beta` CI job: the `drupal/core-*` packages are set as `beta` in `composer.json` but other packages _could_ be resolved as `alpha`.

The table below describes **current** CI jobs of this project (as of 30 May 2024):

|         | D10 version | D10 `minimum-stability` | D11 version | D11 `minimum-stability` |
|---------|-------------|-------------------------|-------------|-------------------------|
| current | `stable`    | `stable`                | `beta`      | `alpha`                 |
| next    | `beta`      | `beta`                  |             |                         |
| canary  | `dev`       | `dev`                   | `dev`       | `dev`                   |

The table below describes **proposed** CI jobs of this project (as of 30 May 2024):

|         | D10 version | D10 `minimum-stability` | D11 version | D11 `minimum-stability` |
|---------|-------------|-------------------------|-------------|-------------------------|
| current | `stable`    | `stable`                | `alpha`*     | `alpha`*                 |
| next    | `beta`      | `beta`                  | `beta`      | `beta`                  |
| canary  | `dev`       | `dev`                   | `dev`       | `dev`                   |

\* These will be changed to `stable` once D11 stable is out.

---

This PR straightens things up by introducing a 
- correct "current" job as `alpha` (per `minimum-stability`), 
- "next" build as `beta`, and 
- "canary" build as `dev`. 

It also resets `drupal/core-*` packages to `alpha` to follow the `minimum-stability` currently set to `alpha`.

Note that as of May 30th 2024, this will install D11 `beta` because D11 is past `alpha`, which is an expected and correct behaviour.

Once D11 is stable, we would only have to:
1. Update `minimum-stability: stable` in `composer.json` 
2. Update `drupal/core-*` versions to `^11` in `composer.json` 
3. Update the version in CI matrix from `alpha` to `stable`.

## Changes

- Added `alpha` job and updated `composer.json` to serve `alpha` for `drupal/core-*` packages to follow the `minimum-stability` currently set  to `alpha`.
- Fixed `dev` version not being correctly installed for `dev` job
- Added installed site status information

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -70,23 +70,30 @@ jobs:
...
-          composer config minimum-stability dev
-          composer --verbose require --no-update drupal/core-recommended:11.0.x-dev
-          composer --verbose require --no-update --dev drupal/core-dev:11.0.x-dev
+          composer config minimum-stability ${{ matrix.drupal-release }}
+          composer config prefer-stable false
+          composer --verbose require --no-update drupal/core-composer-scaffold:^11@${{ matrix.drupal-release }}
+          composer --verbose require --no-update drupal/core-recommended:^11@${{ matrix.drupal-release }}
+          composer --verbose require --no-update --dev drupal/core-dev:^11@${{ matrix.drupal-release }}
+          # Remove the line below once the package is out of `alpha`. 
+          [[ ${{ matrix.drupal-release }} == 'beta' ]] && composer require chi-teck/drupal-code-generator:^4@alpha
```
Comment (AlexSkrypnyk) [inline .github/workflows/ci.yml:81]: this is unfortunate but a required temporary hack to unblock `beta` testing. 

if consumer projects want to use `minimum-stability: beta` before this package is in `beta` - they can explicitly include this package in their `composer.json` using exactly this way

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -70,23 +70,30 @@ jobs:
...
+        run: ./vendor/bin/drush status
+
       - name: Start server
         run: |
           ./vendor/bin/drush runserver "$SIMPLETEST_BASE_URL" &
           until curl -s "$SIMPLETEST_BASE_URL"; do true; done > /dev/null
 
       - name: Run a single unit test to verify the testing setup
-        run: ./vendor/bin/phpunit -c ./web/core $(pwd)/web/core/modules/user/tests/src/Unit/UserAccessControlHandlerTest.php
+        run: ./vendor/bin/phpunit -c ./web/core "$(pwd)/web/core/modules/user/tests/src/Unit/UserAccessControlHandlerTest.php"
```
Comment (AlexSkrypnyk) [inline .github/workflows/ci.yml:96]: just fixing GHA linting issues (using `actionlint`).

File: composer.json [hunk 3]
```diff
@@ -12,15 +12,15 @@
     "require": {
         "composer/installers": "^2.1",
         "cweagans/composer-patches": "^1.7",
-        "drupal/core-composer-scaffold": "^11@beta",
-        "drupal/core-recommended": "^11@beta",
+        "drupal/core-composer-scaffold": "^11@alpha",
```
Comment (AlexSkrypnyk) [inline composer.json:15]: this may seem like a "step back", but it actually makes things less confusing: the versions of `drupal/core-*` packages are inline with the `minimum-stability` currently set to `alpha`.

Comment (normanlolx) [reply composer.json:15]: @AlexSkrypnyk – We will finally change it to stable when stable is out, right?

Comment (AlexSkrypnyk) [reply composer.json:15]: for sure :)

Please see a screenshot of the table from description for reference
Cursor_and_Added_site_information_to_CI_and_`alpha`_job__by_AlexSkrypnyk_·_Pull_Request__671_·_drupal-composer_drupal-project

Comment (normanlolx) [reply composer.json:15]: Alright!

**Review by normanlolx — APPROVED** [review]
Thank you! This looks awesome to me 🤗
